tests/test_integration.py

```diff
@@ -54,9 +50,11 @@ def _do_request(self):
         status = provided_status or self.default_status
         headers = provided_headers or self.default_headers
         body = provided_body or self.default_body
+        content_length = len(body)
         self.send_response(status)
         for header_name, header_value in headers.items():
             self.send_header(header_name, header_value)
+        self.send_header("Content-Length", str(content_length))
```

Comment:
why is this needed?

Reply:
Otherwise the client can't tell whether the request is finished or not and it will hang.
I tried `self.wfile.close()` to close the connection, but this seems to fight with another place that `BaseHTTPHandler` wants to close things, so setting content-length is more expedient. Besides, content-length is standard to include.